packages/firebase_storage/lib/firebase_storage.dart

```diff
@@ -69,6 +69,17 @@ class StorageReference {
     }));
   }
 
+  /// Updates the metadata associated with this [StorageReference].
+  ///
+  /// Returns a [Future] that will complete to the updated [StorageMetadata].
+  Future<StorageMetadata> updateMetadata(StorageMetadata metadata) async {
```

Comment:
This method ignores fields of `metadata` that cannot be set by the public `StorageMetadata` constructor.